### DISCUSSION TITLE
More Blocks Editor

### DISCUSSION THREAD

**Discussion by loui44** [ScratchAddons/ScratchAddons#1396] [Ideas]
Currently when I go to edit my custom block and I want to insert an input somewhere in the middle it won't let me, instead it just adds it on to the end. It would be great if it instead inserted it to just after the currently selected item!

**Comment by ghost** [comment]
> Currently when I go to edit my custom block and I want to insert an input somewhere in the middle it won't let me, instead it just adds it on to the end. It would be great if it instead inserted it to just after the currently selected item!

@loui44 Unfortunately, this is just how Scratch works. It shouldn't really cause any problems. If you really want an addon to be made for this, make a new addon suggestion issue [here](https://github.com/ScratchAddons/ScratchAddons/issues/new?assignees=&labels=type%3A+enhancement%2C+new+addon%2C+scope%3A+addon&template=--new-addon.md&title=).

_It looks like this is a question. If this comment answered your question, please mark it as the answer._

**Comment by TheColaber** [comment]
This could be possible, depending on how making a block works.

**Reply by mxmou** [reply]
Using [these functions](https://github.com/LLK/scratch-blocks/blob/ce41bb947221e4905701f81ba1c615b134b4dd9a/blocks_vertical/procedures.js#L610-L649). The userscript would just need to get a reference to the workspace inside the dialog.

**Reply by TheColaber** [reply]
Awesome! This could totally be a thing then!
